### PR TITLE
DDCE-785 Changed/Added error messages and Regex

### PR DESCRIPTION
**Pull request by meghna2020**
# DDCE-785
**Please review and can be merged now**

## New feature (Changed/Added error message content)

Changed error messages as mentioned in the ticket. Few fields required new error messages so changed regex also.
Got WLU approval.
Added Unit tests.

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings?
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???
- Have you done a manual walkthrough?

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've verified every effort was made to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've verified appropriate tests were included with any code added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've verified commits were squashed and rebased - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/models/RsFormMappings.scala [hunk 1]
```diff
@@ -175,13 +175,26 @@ object RsFormMappings {
       so.length match {
         case x => if (x <= 8) true else false
       }
+    case "schemeOrganiserFields.companyRegistrationNum" =>
```
Comment (RafSobol) [inline app/models/RsFormMappings.scala:178]: collapse that one into the previous case please - they have the same condition (length <= 8)

File: app/models/RsFormMappings.scala [hunk 2]
```diff
@@ -175,13 +175,26 @@ object RsFormMappings {
...
+      so.length match {
+        case x => if (x <= 8) true else false
```
Comment (RafSobol) [inline app/models/RsFormMappings.scala:180]: while at it, would you mind simplifying those lines into 'so.length <= 8'? there's no point matching here.
the method deserves to be changed into something way simpler, but that is a quick starting fix.

File: app/models/RsFormMappings.scala (hunk 1 above)
Comment (meghna2020) [reply app/models/RsFormMappings.scala:176]: Deleted unused code for this.

File: app/models/RsFormMappings.scala (hunk 2 above)
Comment (meghna2020) [reply app/models/RsFormMappings.scala]: Done the changes, replaced the code with so.length <= 8

**Comment by RafSobol** [comment]
looking good (pending adding Welsh & squashing commits)

**Pull request merged**